magento/pwa-studio
# [feature]: ErrorView

## Description

Goal: A generic error view component that handles some basic styling. Should take props allowing for customization, and provide defaults.

Also in this PR:
* Cleanup of unused components
* Refactor of ErrorView and NotFound into single component.
* Fixed some Storybook issues, namely global styles being applied and fonts not being imported.

## Related Issue





Closes [PWA-1032](https://jira.corp.magento.com/browse/PWA-1032).

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Check each of the updated page types and ensure functionality in non-error state.
2. On each of the page types, cause error to be thrown. Simplest way is to block specific data GQL request for that component. Observe that the error view is shown as expected, for that page. Some may have custom messages, others will show default messages.
3. Go to a non-existent route ie https://pr-2936.pwa-venia.com/foo. It should render the error view.

## Screenshots / Screen Captures (if appropriate)

![](https://i.gyazo.com/da3b1e599c3c0bdf345ba99b6cc1065c.png)

![](https://i.gyazo.com/6ae0a312757fe442bc8f1e9fef8ed295.jpg)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
